### PR TITLE
Remove direct use of GlobalContext from package (use an interface)

### PR DESCRIPTION
**Pull request by gabriel**
Removes a direct dependency on GlobalContext for libkbfs package.

This is important for mobile which has issues with the libkb.G global and GlobalContext.

- Replaces libkb.GlobalContext with libkbfs.Context interface
- Replaces access of libkb.G with passed in var (context interface)
- Move construction of context (from GlobalContext) to alternate package (env), since it's used in kbfsfuse, kbfsdokan main packages (and test)
- The impl wraps GlobalContext (implementation hiding) for easy refactoring in the future

In general, we should be using interfaces for library packages like this rather than direct dependencies on something like GlobalContext which itself has a ton of dependencies.

In general we should be deprecating direct use of GlobalContext.

@strib 

cc @maxtaco

### CODE REVIEW

**Comment by gabriel** [comment]
Fun fact... 3rd parties could use this package now without constructing a GlobalContext :)

File: env/context.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+import (
+	"net"
+	"sync"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/go-framed-msgpack-rpc"
+)
+
+// Context is an implementation for libkbfs.Context
```
Comment (strib) [inline env/context.go]: Do you mean `libkb.Context`?  If so, can you please put a `var _ libkb.Context = (*Context)(nil)` somewhere in this file?

Comment (strib) [inline env/context.go]: Oh, nevermind, I see now the new `libkbfs.Context` class now.  Can you please put a `var _ libkbfs.Context = (*Context)(nil)` somewhere then?

**Comment by strib** [comment]
Nice, this looks good to me, just the one suggestion.  Thanks!

**Pull request merged**